### DISCUSSION TITLE
MotionSensor_v2 Op "does not work in an iframe"?

### DISCUSSION THREAD

**Discussion by jimimased** [cables-gl/cables_docs#384] [Answers & Questions] [answered]
I am making an A.R project. Point mobile at laptop browser, viewed through mobile, see around the 3d objects floating on top of the laptop keyboard & alter the perspective of the laptop browser simultaneously. Im trying the motionsensor2 patch, such a great feature!  

Using [Kirell Benzi](https://www.youtube.com/c/KirellBenzi1)s example i can alter the cubes position on the laptop browser via the slider on the mobile & socket cluster works perfectly on iOS 15.6 on both chrome & safari.

BUT I get no xyz orientation from motionsensor2 ? the cube in Kirells patch doesnt alter on mobile either?

When I look at the motion sensor patch it says  "Warning: MotionSensor does not work in an iframe, open the patch without an iframe to get it to work" 

how do I fix this issue & get xyz orientation from motionsensor working? 

Thank you!

**Comment by steam0r** [comment]
hey,

warning clearly states the problem :) don't run the patch in an iframe (for example on cables.gl), use export to host things on your own server.

we have to run patches on cables.gl in iframes for security reasons.

**Reply by FifthChat** [reply]
> hey,
> 
> warning clearly states the problem :) don't run the patch in an iframe (for example on cables.gl), use export to host things on your own server.
> 
> we have to run patches on cables.gl in iframes for security reasons.

Hi，I export path to run on my server, and motion sensor is also unavailable.

**Comment by pandrr** [comment, marked as answer]
i made the example more clear, added a play button, because you need a user interaction to start. 

here you see it working, exported
https://preview.undev.studio/temp/motion

**Reply by jimimased** [reply]
Got it Fantastic.

Thank you Pandrr

**Reply by jshea2** [reply]
Hey @pandrr, Thanks for this! Do you have any resources on how to deploy a cables project to unidev.studio?

**Reply by pandrr** [reply]
this is a private webspace, you can export any patch and upload it to most providers, you need to have file access to a webserver
